src/Test/test_model.jl

```diff
@@ -0,0 +1,1019 @@
+    test_model_is_valid(model::MOI.ModelLike, config::Config)
+
+Test various parts of `MOI.is_valid`.
+
+Taken from https://github.com/jump-dev/MathOptInterfaceUtilities.jl/issues/41
+"""
+function test_model_is_valid(model::MOI.ModelLike, config::Config)
+    @requires MOI.supports_incremental_interface(model, false)
+    v = MOI.add_variables(model, 2)
+    @test MOI.is_valid(model, v[1])
+    @test MOI.is_valid(model, v[2])
+    x = MOI.add_variable(model)
+    @test MOI.is_valid(model, x)
+    if _supports(config, MOI.delete)
+        MOI.delete(model, x)
+        @test !MOI.is_valid(model, x)
+    end
+    cf = MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.([1.0, 1.0], v), 0.0)
+    @test MOI.supports_constraint(model, typeof(cf), MOI.LessThan{Float64})
+    c = MOI.add_constraint(model, cf, MOI.LessThan(1.0))
+    @test MOI.is_valid(model, c)
+    @test !MOI.is_valid(
+        model,
+        MOI.ConstraintIndex{
+            MOI.ScalarAffineFunction{Float32},
+            MOI.LessThan{Float32},
+        }(
+            1,
+        ),
+    )
+    @test !MOI.is_valid(
+        model,
+        MOI.ConstraintIndex{
+            MOI.ScalarAffineFunction{Float32},
+            MOI.LessThan{Float64},
+        }(
+            1,
+        ),
+    )
+    @test !MOI.is_valid(
+        model,
+        MOI.ConstraintIndex{
+            MOI.ScalarAffineFunction{Float64},
+            MOI.LessThan{Float32},
+        }(
+            1,
+        ),
+    )
+    @test !MOI.is_valid(
+        model,
+        MOI.ConstraintIndex{
+            MOI.VectorQuadraticFunction{Float64},
+            MOI.SecondOrderCone,
+        }(
+            1,
+        ),
+    )
+    return
+end
+
+"""
+    test_model_empty(model::MOI.ModelLike, ::Config)
+
+Test `MOI.empty` and `MOI.is_empty!`.
+"""
+function test_model_empty(model::MOI.ModelLike, ::Config)
+    @test MOI.is_empty(model)
+    x = MOI.add_variable(model)
+    @test !MOI.is_empty(model)
+    MOI.empty!(model)
+    @test MOI.is_empty(model)
+    return
+end
+
+"""
+    test_model_copy_to_UnsupportedConstraint(
+        model::MOI.ModelLike,
+        ::Config,
+    )
+
+Test `MOI.copy_to` when a constraint is unsupported.
+"""
+function test_model_copy_to_UnsupportedConstraint(
+    model::MOI.ModelLike,
+    ::Config,
+)
+    @test !MOI.supports_constraint(
+        model,
+        MOI.SingleVariable,
+        UnknownScalarSet{Float64},
+    )
+    @test_throws(
+        MOI.UnsupportedConstraint,
+        MOI.copy_to(model, BadConstraintModel()),
+    )
+    return
+end
+
+"""
+    test_model_copy_to_UnsupportedAttribute(
+        model::MOI.ModelLike,
+        ::Config,
+    )
+
+Test `MOI.copy_to` when an attribute is unsupported.
+"""
+function test_model_copy_to_UnsupportedAttribute(model::MOI.ModelLike, ::Config)
+    # ModelAttribute
+    @test !MOI.supports(model, UnknownModelAttribute())
+    @test_throws(
+        MOI.UnsupportedAttribute,
+        MOI.copy_to(model, BadModelAttributeModel()),
+    )
```

Comment:
@blegat I found test GLPK fails! It was trying to access `ObjectiveFunction` attribute, despite the fact that the model does not list in in the `ListOfModelAttributesSet`. Hurrah for full testing from modellike.

Reply:
```Julia
julia> @testset "MOI" begin
           MOI.Test.runtests(
               GLPK.Optimizer(),
               MOI.Test.Config(),
               exclude = [
                   # GLPK treats this one as "INVALID_MODEL"
                   "test_constraint_ZeroOne_bounds_3",
               ]
           )
       end
Test Summary: | Pass  Total
MOI           |  476    476
[...]
```